corehq/apps/geospatial/views.py

```diff
@@ -1,8 +1,67 @@
-from django.http import HttpResponseRedirect
+import json
+import jsonschema
+
+from django.http import HttpResponseRedirect, Http404, HttpResponseBadRequest
+from django.utils.decorators import method_decorator
 from django.utils.translation import gettext_lazy as _
+from dimagi.utils.web import json_response
 
+from corehq import toggles
+from corehq.apps.domain.views.base import BaseDomainView
 from corehq.apps.geospatial.reports import CaseManagementMap
+from .const import POLYGON_COLLECTION_GEOJSON_SCHEMA
+from .models import GeoPolygon
 
 
 def geospatial_default(request, *args, **kwargs):
     return HttpResponseRedirect(CaseManagementMap.get_url(*args, **kwargs))
+
+
+class GeoPolygonView(BaseDomainView):
+    urlname = 'geo_polygon'
+
+    @method_decorator(toggles.GEOSPATIAL.required_decorator())
+    def dispatch(self, request, *args, **kwargs):
+        return super(GeoPolygonView, self).dispatch(request, *args, **kwargs)
+
+    def get(self, request, *args, **kwargs):
+        try:
+            polygon_id = int(request.GET.get('polygon_id', None))
+        except TypeError:
+            raise Http404()
+        try:
+            polygon = GeoPolygon.objects.get(pk=polygon_id)
+            assert polygon.domain == self.domain
+        except (GeoPolygon.DoesNotExist, AssertionError):
+            raise Http404()
+        return json_response(polygon.geo_json)
+
+    def post(self, request, *args, **kwargs):
+        try:
+            geo_json = json.loads(request.body).get('geo_json', None)
+        except json.decoder.JSONDecodeError:
+            raise HttpResponseBadRequest(
+                'POST Body must be a valid json in {"geo_json": <geo_json>} format'
+            )
+
+        if not geo_json:
+            raise HttpResponseBadRequest('Empty geo_json POST field')
+
+        try:
+            jsonschema.validate(geo_json, POLYGON_COLLECTION_GEOJSON_SCHEMA)
```

Comment:
just an FYI, this can be added to the model field as a validator. e.g. https://github.com/dimagi/commcare-hq/blob/f5e52a39e7a6ba38240844e251881cdb718b944b/corehq/apps/registry/models.py#L75